references/detection/transforms.py

```diff
@@ -591,3 +591,80 @@ def forward(
     def __repr__(self) -> str:
         s = f"{self.__class__.__name__}(blending={self.blending}, resize_interpolation={self.resize_interpolation})"
         return s
+
+
+class Mosaic(nn.Module):
+    def __init__(self, min_frac: float = 0.25, max_frac: float = 0.75, size_limit=10) -> None:
+        super().__init__()
+        self.min_frac = min_frac
+        self.max_frac = max_frac
+        self.size_limit = size_limit
+
+    def forward(self, images, targets):
+        """
+        images : torch.Tensor of NWHC type
+        targets: list[torch.Tensor]; bounding boxes in xyxy format.
+        """
+
+        # implementation  is heavily inspired from this colab notebook
+        sx, sy = images.shape[-2:]
+
+        num_channels = images.shape[-3]
```

Comment:
Sorry for early poke but I could spot a error so pinging.

I think this Doesn't seem to be the case for NHWC image? Are sure?

Reply:
Yeah, right..

Forgot to update the doc strings, 
Earlier implementation was for channels last, but i think in Torchvision we are mainly working with channels first data, so
changes for this were made in https://github.com/pytorch/vision/pull/6534/commits/ced3ac456bb516fba2320028026ff82cf265fa21,

Reply:
Sorry for poor code quality. Still figuring out ways to have multiple batches without using for loops, will update you all once done.